server/src/main/java/org/opensearch/search/startree/filter/provider/StarTreeFilterProvider.java

```diff
@@ -156,4 +163,176 @@ public StarTreeFilter getFilter(SearchContext context, QueryBuilder rawFilter, C
 
     }
 
+    class BoolStarTreeFilterProvider implements StarTreeFilterProvider {
+        @Override
+        public StarTreeFilter getFilter(SearchContext context, QueryBuilder rawFilter, CompositeDataCubeFieldType compositeFieldType)
+            throws IOException {
+            BoolQueryBuilder boolQueryBuilder = (BoolQueryBuilder) rawFilter;
+            if (!boolQueryBuilder.hasClauses()) {
+                return null;
+            }
+
+            if (boolQueryBuilder.must().isEmpty() == false) {
+                return processMustClauses(boolQueryBuilder.must(), context, compositeFieldType);
```

Comment:
Lets add filter clauses + must clauses together always ? constraints are same as scoring is anyways not applicable to star tree index.

Reply:
Done